src/shapes/Object/Object.ts

```diff
@@ -820,10 +801,12 @@ export class FabricObject<
   }
 
   /**
-   * Sets object's properties from options
+   * Sets object's properties from options, for class constructor only.
+   * Needs to be overridden for different defaults.
+   * @protected
    * @param {Object} [options] Options object
    */
-  setOptions(options: Record<string, any> = {}) {
+  protected setOptions(options: Record<string, any> = {}) {
     this._setOptions(options);
   }
```

Comment:
do we really need this alias?
It is called only once in fabric from this contrusctor

Reply:
and `_setOptions` is an alias for `set` so that can go away as well.
too many methods that do the same shit

Reply:
not this should go because no strange stuf in the constructor. It will go when we move the defaults